backend/functions/src/resources/user-profile/user-profile.controller.ts

```diff
@@ -0,0 +1,18 @@
+import { RequestHandler } from 'express';
+import { UserProfile } from '.';
+import { getUserProfileService, patchUserProfileService } from '../../services/user-profile.service';
+
+export const getUserProfile: RequestHandler = async (req, res) => {
+  if (typeof req.params.id === 'string') {
+    const profile = await getUserProfileService(req.params.id);
+    res.send(profile);
+  } else res.send('Error');
+};
+
+export const patchUserProfile: RequestHandler = async (req, res) => {
+  if (typeof req.params.id === 'string') {
+    const userProfile: UserProfile = req.body.profile;
```

Comment:
The `req.body` object should be the `profile` object, rather than the `body` having a `profile` field. 

That is, the contents of `req.body` should be the profile fields.